satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -574,6 +607,13 @@ def __init__(self, filename, filename_info, filetype_info,  # noqa: D417
             self._get_acq_time_uncached
         )
 
+    def _get_projection_longitude(self, filename_info):
+        """Read projection longitude from filename as it is not provided in the file."""
+        if "." in str(filename_info["projection_longitude"]):
```

Comment:
I just learned from an AI that you can use `filename_info["projection_longitude"].is_integer()` to check if a number has zero digits

Reply:
nice function! However, it does not serve the purpose of the code above which needs to distinguish e.g. 57.0 from 5700 in mviri filenames version 1 and 2